lib/iris/coords.py

```diff
@@ -1720,11 +1730,7 @@ def points(self, points):
         # Ensure points has an ndmin of 1 and is either a numpy or lazy array.
         # This will avoid Scalar coords with points of shape () rather
         # than the desired (1,).
-        if is_lazy_data(points):
-            if points.shape == ():
-                points = points.reshape(1,)
-        else:
-            points = self._sanitise_array(points, 1)
+        points = self._sanitise_array(points, 1)
```

Comment:
It feels like you should go the whole hog and integrate these changes into `DimCoord` too. But, as per my previous comment, I'd prefer if that were done in another PR.